File: src/CSharp/223.RectangleArea.cs

using System;
using System.Collections.Generic;

namespace PeefyLeetCode.RectangleArea
{
public class Solution {
    public int ComputeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
        if (E > C || A > G)
            return (C - A) * (D - B) + (G - E) * (H - F);
        if (F > D || H < B)
            return (C - A) * (D - B) + (G - E) * (H - F);
        int length = Math.Min(C, G) - Math.Max(A, E);
        int width = Math.Min(D, H) - Math.Max(B, F);
        return (C - A) * (D - B) + (G - E) * (H - F) - length * width;
    }
}
}

File: src/Python/223.RectangleArea.py

class Solution:
    def computeArea(self, A, B, C, D, E, F, G, H):
        if E > C or A > G: 
            return (C - A) * (D - B) + (G - E) * (H - F)
        if F > D or H < B: 
            return (C - A) * (D - B) + (G - E) * (H - F)
        length = min(C, G) - max(A, E)
        width = min(D, H) - max(B, F)
        return (C - A) * (D - B) + (G - E) * (H - F) - length * width


if __name__ == "__main__":
    solution = Solution()
    print(solution.computeArea(-3, 0, 3, 4, 0, -1, 9, 2))

